uefi-test-runner/src/proto/misc.rs

```diff
@@ -0,0 +1,80 @@
+use uefi::prelude::*;
+use uefi::proto::misc::{ResetNotification, Timestamp};
+use uefi::table::runtime;
+
+///
+/// you may see those log, it's nothing just for your computer firmware does not support the new UEFI feature.
+///
+/// ```sh
+/// [ INFO]: uefi-test-runner\src\proto\misc.rs@012: Running loaded Timestamp Protocol test
+/// [ WARN]: uefi-test-runner\src\proto\misc.rs@026: Failed to open Timestamp Protocol: Error { status: UNSUPPORTED, data: () }
+/// [ INFO]: uefi-test-runner\src\proto\misc.rs@033: Running loaded ResetNotification protocol test
+/// [ WARN]: uefi-test-runner\src\proto\misc.rs@068: Failed to open ResetNotification Protocol: Error { status: UNSUPPORTED, data: () }
+/// ```
+pub fn test(image: Handle, bt: &BootServices) {
+    test_timestamp(image, bt);
+    test_reset_notification(image, bt);
+}
+
+pub fn test_timestamp(image: Handle, bt: &BootServices) {
+    info!("Running loaded Timestamp Protocol test");
+
+    let result = bt
+        .open_protocol_exclusive::<Timestamp>(image);
+
+    match result {
+        Ok(timestamp_proto) => {
+            let timestamp = timestamp_proto.get_timestamp();
+            info!("Timestamp Protocol's timestamp: {:?}", timestamp);
+
+            let properties = timestamp_proto.get_properties();
+            info!("Timestamp Protocol's properties: {:?}", properties);
+        }
+        Err(err) => {
+            warn!("Failed to open Timestamp Protocol: {:?}", err);
+        }
+    }
+}
+
+
+pub fn test_reset_notification(image: Handle, bt: &BootServices) {
```

Comment:
nit: This only tests to set hook but not that the hook is actually called. Can we fix/add that?